src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -481,38 +477,100 @@ Namespace Microsoft.CodeAnalysis.Operations
                 Dim userDefinedConversion = DirectCast(boundConversion.Operand, BoundUserDefinedConversion)
-                Return (Operand:=userDefinedConversion.Operand, Method:=userDefinedConversion.Call.Method)
+                Return userDefinedConversion.Operand
+            Else
+                Return boundConversion.Operand
+            End If
+        End Function
+
+        Private Function CreateNestedConversionOperand(operand As BoundExpression, conversionSyntax As SyntaxNode) As Lazy(Of IOperation)
+            Return New Lazy(Of IOperation)(Function() CreateNestedConversionOperandCore(operand))
+        End Function
+
+        Private Function CreateNestedConversionOperandCore(operand As BoundExpression) As IOperation
+            Select Case operand.Kind
+                Case BoundKind.Parenthesized
+                    Dim boundParenthesized = DirectCast(operand, BoundParenthesized)
+                    ' Because we're in a delegate creation scenario, the underlying expression actually doesn't have a type.
+                    Dim operandOperation = CreateNestedConversionOperandCore(boundParenthesized.Expression)
+                    Return New ParenthesizedExpression(
+                        operandOperation,
+                        _semanticModel,
+                        boundParenthesized.Syntax,
+                        operandOperation.Type,
+                        ConvertToOptional(boundParenthesized.ConstantValueOpt),
+                        boundParenthesized.WasCompilerGenerated)
+                Case BoundKind.Conversion, BoundKind.DirectCast, BoundKind.TryCast
+                    ' This is the artificial conversion. Erase it from the tree.
+                    Return CreateConversionOperand(DirectCast(operand, BoundConversionOrCast).Operand)
+                Case BoundKind.DelegateCreationExpression
+                    Return CreateBoundDelegateCreationExpressionChildOperation(DirectCast(operand, BoundDelegateCreationExpression))
+                Case Else
+                    Throw ExceptionUtilities.UnexpectedValue(operand.Kind)
+            End Select
+        End Function
+
+        Private Function CreateConversionOperand(operand As BoundExpression) As IOperation
+            If operand.Kind = BoundKind.DelegateCreationExpression Then
+                ' If the child is a BoundDelegateCreationExpression, we don't want to generate a nested IDelegateCreationExpression.
+                ' So, the operand for the conversion will be the child of the BoundDelegateCreationExpression.
+                ' We see this in this syntax: Dim x = New Action(AddressOf M2)
+                ' This should be semantically equivalent to: Dim x = AddressOf M2
+                ' However, if we didn't fix this up, we would have nested IDelegateCreationExpressions here for the former case.
+                Return CreateBoundDelegateCreationExpressionChildOperation(DirectCast(operand, BoundDelegateCreationExpression))
```

Comment:
>Return CreateBoundDelegateCreationExpressionChildOperation(DirectCast(operand, BoundDelegateCreationExpression)) [](start = 16, length = 112)

It feels fragile that we are making assumption that IsDelegateCreation will be set to true in this case.